core/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -478,8 +485,24 @@ public ActionRequestValidationException validate() {
         if (requests.isEmpty()) {
             validationException = addValidationError("no requests added", validationException);
         }
-        for (int i = 0; i < requests.size(); i++) {
-            ActionRequestValidationException ex = requests.get(i).validate();
+        for (ActionRequest request : requests) {
+            // We first check if refresh has been set
+            if (request instanceof DeleteRequest) {
+                if (((DeleteRequest)request).refresh()) {
+                    validationException = addValidationError("Refresh is not supported for DeleteRequest when added to a BulkRequest", validationException);
```

Comment:
maybe be nice and add that you reuse the bulk request.  So "refresh is not supported on an item request, set the refresh flag on the BulkRequest instead".